### PR TITLE
dependencyjs complete implementation

### PR DESCRIPTION
**Pull request by ManishKumawat**
This include all the changes related to registration, resolution and configuration reading.

### CODE REVIEW

**Comment by ManishKumawat** [comment]
will create new PR

File: src/ConfigurationReader.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                        map = require(registrationSection.mapProperty.sourceInfo)[registrationSection.mapTo];
+                    }
+                    else {
+                        map = require(path.join(folderConfig.appDirectory + registrationSection.mapProperty.sourceInfo))[registrationSection.mapTo];
+                    }
+
+                    ComponentRegistry.getInstance().register(base, new map(), registrationSection.resolver);
+                }
+                catch (ex) {
+                    console.log(registrationSection.mapTo + ": object cant be registered due to some issues. Exception: " + ex);
```
Comment (rakeshsawan) [inline src/ConfigurationReader.ts]: we shoudl throw an exception in this case right?

File: src/ConfigurationReader.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        return true;
+    }
+
+    private registerDependencies(configuration: ConfigurationSection, folderConfig:Config) {
+        if (configuration != null) {
+            configuration.registrationSections.forEach((registrationSection: RegisterSection) => {
+                try {
+                    let base: any;
+                    let map: any;
```
Comment (rakeshsawan) [inline src/ConfigurationReader.ts]: could we have a separate method what will take care of determining base and map?

File: src/ConfigurationReader.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                    else {
+                        configSections.registrationSections.push(registrationData);
+                    }
+                }
+                else {
+                    console.log("No configuration found in registration section : " + fileName);
+                }
+            }
+            catch (ex) {
+                console.log("There is some issues with configuration file :" + fileName);
```
Comment (rakeshsawan) [inline src/ConfigurationReader.ts]: should throw an exception.

File: src/ConfigurationReader.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        let configSections: ConfigurationSection = new ConfigurationSection();
+        configSections.registrationSections = new Array<RegisterSection>();
+
+        //read all component files
+        let folderPath: string = path.join(config.appDirectory, config.configXmlFolder);
+        let fileNames: Array<string> = fs.readdirSync(folderPath);
+        fileNames.forEach(fileName => {
+            try {
+                let section: any = fs.readFileSync(folderPath + "/" + fileName);
```
Comment (rakeshsawan) [inline src/ConfigurationReader.ts:47]: This could be a separate method.

File: gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+};
+let compile = [
+  'index.ts',
+  './src/**/*.ts',
+  './test/**/*.ts',
+  './src/**/*Spec.ts',
+  '!./src/**/trash/**'
+];
+
+gulp.task("copy-html", function () {
```
Comment (rakeshsawan) [inline gulpfile.js]: what is the purpose of copy-html target? may be a rename is required or we can just remove it.

File: src/ConfigurationReader.ts (hunk 1 above)
Comment (ManishKumawat) [reply src/ConfigurationReader.ts]: Okay. I will do it

File: gulpfile.js (hunk 5 above)
Comment (ManishKumawat) [reply gulpfile.js]: Sure. i will remove it

**Pull request closed without merging**